app/src/org/commcare/dalvik/activities/LoginActivity.java

```diff
@@ -74,11 +73,35 @@
     
     @UiElement(R.id.str_version)
     TextView versionDisplay;
+
+    @UiElement(R.id.login_button)
+    Button loginButton;
     
     public static final int TASK_KEY_EXCHANGE = 1;
     
     SqlStorage<UserKeyRecord> storage;
-    
+
+    private int editTextColor;
+    private View.OnKeyListener l;
+    private final TextWatcher textWatcher = new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+                setLoginBoxesColor(editTextColor, false);

```

Comment:
i'd pull all of these into a helper method in the activity called "setStyleDefault" or something similar to make it more clear what exactly each element is doing and make it triggerable from outside of this context as well.